llvm/include/llvm/BinaryFormat/DXContainer.h

```diff
@@ -546,13 +560,75 @@ struct ProgramSignatureElement {
 static_assert(sizeof(ProgramSignatureElement) == 32,
               "ProgramSignatureElement is misaligned");
 
+struct RootConstants {
+  uint32_t ShaderRegister = 0;
+  uint32_t RegisterSpace = 0;
+  uint32_t Num32BitValues = 0;
+
+  RootConstants() = default;
+
+  void swapBytes() {
+    sys::swapByteOrder(ShaderRegister);
+    sys::swapByteOrder(RegisterSpace);
+    sys::swapByteOrder(Num32BitValues);
+  }
+};
+
+struct RootParameter {
+  dxbc::RootParameterType ParameterType;
+  union {
+    dxbc::RootConstants Constants;
+  };
+  dxbc::ShaderVisibilityFlag ShaderVisibility;
+
+  RootParameter() {
+    Constants = RootConstants();
+    ParameterType = dxbc::RootParameterType::Empty;
+    ShaderVisibility = dxbc::ShaderVisibilityFlag::Empty;
+  }
+
+  void swapBytes() {
+
+    sys::swapByteOrder(ParameterType);
```

Comment:
nit, but these things stand out to me and I find it hard to ignore them :)

```suggestion
  void swapBytes() {
    sys::swapByteOrder(ParameterType);
```